### PR TITLE
BROKERS: GET All Teachers

### PR DESCRIPTION
**Pull request by coeur85**

### CODE REVIEW

File: OtripleS.Portal.Web/.vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{
```
Comment (mtoncic) [inline OtripleS.Portal.Web/.vscode/launch.json:1]: I guess this was not intentionally added

File: OtripleS.Portal.Web/.vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (mtoncic) [inline OtripleS.Portal.Web/.vscode/tasks.json:1]: I guess this was not intentionally added

File: OtripleS.Portal.Web/Brokers/API/ApiBroker.Teachers.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+namespace OtripleS.Portal.Web.Brokers.API
+{
+      public partial class ApiBroker
+      {
+            private const string TeachersRelativeUrl = "api/teachers";
+
+            public async ValueTask<List<Teacher>> GetAllTeachers() =>
+                await this.GetAsync<List<Teacher>>(TeachersRelativeUrl);
+      }
+}
```
Comment (mtoncic) [inline OtripleS.Portal.Web/Brokers/API/ApiBroker.Teachers.cs:19]: Add one more line at the end. Also in other files where `new line at the end of the file` is missing.

**Review by mtoncic — CHANGES_REQUESTED** [review]

File: OtripleS.Portal.Web/.vscode/launch.json (hunk 1 above)
Comment (coeur85) [reply OtripleS.Portal.Web/.vscode/launch.json:1]: true story, t'm using vs code on mac

File: OtripleS.Portal.Web/Models/Teachers/Teacher.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+namespace OtripleS.Portal.Web.Models.Teachers
+{
+    public class Teacher
+    {
+        
+    }
```
Comment (hassanhabib) [inline OtripleS.Portal.Web/Models/Teachers/Teacher.cs:6]: Where's the properties on the teacher model?
Also don't forget to add a file header

Comment (coeur85) [reply OtripleS.Portal.Web/Models/Teachers/Teacher.cs:6]: I've been down this road in my previous PR
until we finally agreed to leave it empty

Please check this out, and tell me what's next

https://github.com/hassanhabib/OtripleS.Portal/pull/17

File: OtripleS.Portal.Web/.vscode/tasks.json (hunk 2 above)
Comment (hassanhabib) [reply OtripleS.Portal.Web/.vscode/tasks.json:1]: @coeur85 same thing let's remove this file as well please.

File: OtripleS.Portal.Web/Brokers/API/IApiBroker.Teachers.cs [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OtripleS.Portal.Web.Models.Teachers;
+
+namespace OtripleS.Portal.Web.Brokers.API
+{
+      public partial interface IApiBroker
+      {
+            ValueTask<List<Teacher>> GetAllTeachers();
```
[the remark this answers is not in the document]
Comment (hassanhabib) [reply OtripleS.Portal.Web/Brokers/API/IApiBroker.Teachers.cs:14]: @coeur85 create a quick PR to fix that in the `Student` broker. quick point.

File: OtripleS.Portal.Web/.vscode/tasks.json (hunk 2 above)
Comment (coeur85) [reply OtripleS.Portal.Web/.vscode/tasks.json]: done

File: OtripleS.Portal.Web/Brokers/API/IApiBroker.Teachers.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (coeur85) [reply OtripleS.Portal.Web/Brokers/API/IApiBroker.Teachers.cs]: i've updated the teachers broker

File: OtripleS.Portal.Web/Models/Teachers/TeacherStatus.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace OtripleS.Portal.Web.Models.Teachers
+{
+      public enum TeacherStatus
+      {
+            Working,
+            Fired,
+            Died,
+            Resigned
+
```
Comment (mtoncic) [inline OtripleS.Portal.Web/Models/Teachers/TeacherStatus.cs:14]: Remove this extra empty line

File: OtripleS.Portal.Web/Models/Teachers/TeacherStatus.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
+namespace OtripleS.Portal.Web.Models.Teachers
+{
+      public enum TeacherStatus
+      {
+            Working,
```
Comment (mtoncic) [inline OtripleS.Portal.Web/Models/Teachers/TeacherStatus.cs:10]: Where did you find this `TeacherStatus` enum values?

File: OtripleS.Portal.Web/Models/Teachers/Teacher.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
+// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
```
Comment (mtoncic) [inline OtripleS.Portal.Web/Models/Teachers/Teacher.cs:5]: Remove this extra line

File: OtripleS.Portal.Web/Models/Teachers/TeacherGender.cs [hunk 9]
```diff
@@ -0,0 +1,17 @@
+// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE AS LONG AS SOFTWARE FUNDS ARE DONATED TO THE POOR
+// ---------------------------------------------------------------
+
```
Comment (mtoncic) [inline OtripleS.Portal.Web/Models/Teachers/TeacherGender.cs:5]: Remove this extra line

**Pull request merged**